pkg/cloud/services/compute/instance.go

```diff
@@ -194,54 +175,31 @@ func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infr
 			}
 		}
 
-		portsList = append(portsList, servers.Network{
+		portList = append(portList, servers.Network{
 			Port: port.ID,
 		})
-
-		if i.Trunk {
-			allPages, err := trunks.List(is.networkClient, trunks.ListOpts{
-				Name:   i.Name,
-				PortID: port.ID,
-			}).AllPages()
-			if err != nil {
-				return nil, fmt.Errorf("searching for existing trunk for server err: %v", err)
-			}
-			trunkList, err := trunks.ExtractTrunks(allPages)
-			if err != nil {
-				return nil, fmt.Errorf("searching for existing trunk for server err: %v", err)
-			}
-			var trunk trunks.Trunk
-			if len(trunkList) == 0 {
-				// create trunk with the previous port as parent
-				trunkCreateOpts := trunks.CreateOpts{
-					Name:   i.Name,
-					PortID: port.ID,
-				}
-				newTrunk, err := trunks.Create(is.networkClient, trunkCreateOpts).Extract()
-				if err != nil {
-					return nil, fmt.Errorf("create trunk for server err: %v", err)
-				}
-				trunk = *newTrunk
-			} else {
-				trunk = trunkList[0]
-			}
-
-			_, err = attributestags.ReplaceAll(is.networkClient, "trunks", trunk.ID, attributestags.ReplaceAllOpts{
-				Tags: i.Tags,
-			}).Extract()
-			if err != nil {
-				return nil, fmt.Errorf("tagging trunk for server err: %v", err)
-			}
-		}
 	}
 
 	if i.Subnet != "" && accessIPv4 == "" {
-		if errd := deletePorts(is, portsList); errd != nil {
-			return nil, fmt.Errorf("no ports with fixed IPs found on Subnet %q: error cleaning up ports: %v", i.Subnet, errd)
+		if errd := deletePorts(is, object, portList); errd != nil {
+			return nil, errd
 		}
 		return nil, fmt.Errorf("no ports with fixed IPs found on Subnet %q", i.Subnet)
 	}
 
+	instance, err := createServer(is, object, portList, accessIPv4, i)
```

Comment:
Does it really make sense to split this func? I don't see a good semantic difference between createInstance and createServer and createServer is only ever called by createInstance

Reply:
Thanks question. That is What I considered. If `createInstance` is "business logic" which means call `getOrCreatePort`, and if trunk true, call `getOrCreateTrunk` then `createServer`, current is better. Otherwise, we should not split function, `createInstance` includes `createServer` as main logic. I got your suggestion and will not split function.